azurerm/internal/services/postgres/postgresql_flexible_server_configuration_test.go

```diff
@@ -0,0 +1,146 @@
+			),
+		},
+	})
+}
+
+func TestAccFlexibleServerConfiguration_updateApplicationName(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_postgresql_flexible_server_configuration", "test")
+	r := PostgresqlFlexibleServerConfigurationResource{}
+	name := "application_name"
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.template(data, name, "Test APP before"),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("name").HasValue(name),
+				check.That(data.ResourceName).Key("value").HasValue("Test APP before"),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.template(data, name, "Test APP after"),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("name").HasValue(name),
+				check.That(data.ResourceName).Key("value").HasValue("Test APP after"),
+			),
+		},
+	})
+}
+
+func (r PostgresqlFlexibleServerConfigurationResource) checkReset(configurationName string) acceptance.ClientCheckFunc {
+	return func(ctx context.Context, clients *clients.Client, state *pluginsdk.InstanceState) error {
+		id, err := parse.FlexibleServerID(state.Attributes["id"])
+		if err != nil {
+			return err
+		}
+
+		resp, err := clients.Postgres.FlexibleServersConfigurationsClient.Get(ctx, id.ResourceGroup, id.Name, configurationName)
+		if err != nil {
+			if utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Bad: Azure Postgresql Flexible Server configuration %q (server %q resource group: %q) does not exist", configurationName, id.Name, id.ResourceGroup)
+			}
+			return fmt.Errorf("Bad: Get on postgresqlConfigurationsClient: %+v", err)
+		}
+
+		actualValue := *resp.Value
+		defaultValue := *resp.DefaultValue
+
+		if defaultValue != actualValue {
+			return fmt.Errorf("Azure Postgresql Flexible Server configuration wasn't set to the default value. Expected '%s' - got '%s': \n%+v", defaultValue, actualValue, resp)
+		}
+
+		return nil
+	}
+}
+
+// Helper functions for verification
+func (t PostgresqlFlexibleServerConfigurationResource) Exists(ctx context.Context, clients *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := parse.FlexibleServerConfigurationID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := clients.Postgres.FlexibleServersConfigurationsClient.Get(ctx, id.ResourceGroup, id.FlexibleServerName, id.ConfigurationName)
+	if err != nil {
+		return nil, fmt.Errorf("reading Postgresql Configuration (%s): %+v", id.String(), err)
+	}
+
+	return utils.Bool(resp.ID != nil), nil
+}
+
+func (PostgresqlFlexibleServerConfigurationResource) empty(data acceptance.TestData) string {
+	return PostgresqlFlexibleServerResource{}.basic(data)
+}
+
+func (r PostgresqlFlexibleServerConfigurationResource) template(data acceptance.TestData, name, value string) string {
+	return fmt.Sprintf(`
```

Comment:
Could we rename these two funcs - as a convention we generally use `template()` for the required parent resource configs and something like `basic()` for the resource config being tested.

Reply:
renamed